src/plugins/intel_npu/src/plugin/npuw/weights_bank.hpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+
+#include "openvino/openvino.hpp"
+#include "openvino/runtime/itensor.hpp"
+#include "openvino/runtime/iremote_context.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/intel_npu/level_zero/level_zero.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "plugin.hpp"
+
+namespace ov {
+namespace npuw {
+namespace weights_bank {
+
+class WeightBank {
+public:
+    WeightBank(const std::string& device_name, const std::shared_ptr<const ov::IPlugin>& plugin) : m_device_name(device_name), m_plugin(plugin) {
+        if (m_device_name != "CPU" && m_device_name != "NPU") {
+            OPENVINO_THROW("NPUW doesn't support ", m_device_name, " device for weights sharing!");
+        }
+        if (m_device_name == "NPU" && !m_remote_ctx) {
+            m_remote_ctx = m_plugin->get_core()->get_default_context("NPU")._ptr;
```

Comment:
Not sure if it works properly for NPU. Ideally that should have been `LevelZeroContext`